src/v2/ion/uiSchemaLabelTransformer.js

```diff
@@ -158,15 +163,42 @@ const updateLabelForUiSchema = (remediation, uiSchema) => {
 
 };
 
+const getMessage = (message) => {
+  if (message.i18n && message.i18n.key) {
+    const i18nKey = message.i18n.key;
+    if (Bundles.login[i18nKey]) {
+      Logger.info(`Override messages using i18n key ${i18nKey}`);
+      // expect user config i18n properly.
+      // e.g. the i18n value shall have placeholders like `{0}`, when params is not empty.
+      return loc(i18nKey, 'login', message.i18n.params || []);
+    }
+  }
+  return message.message;
+};
+
 const uiSchemaLabelTransformer = (transformedResp) => {
+  // Try to override label using i18n value
   if (Array.isArray(transformedResp.remediations)) {
     transformedResp.remediations
       .filter(remediation => Array.isArray(remediation.uiSchema) && remediation.uiSchema.length)
       .forEach(remediation => {
         remediation.uiSchema.forEach(uiSchema => updateLabelForUiSchema(remediation, uiSchema));
       });
   }
+
+  // Try to override `messages` using i18n value.
+  // 1. This is only handling top level `messages` object when response status is 200.
+  // 2. See `IonResponseHelper.js` where handle `messages` object when none 200 response.
+  // 3. Handling `messages` in remediation forms on 200 response is not considered yet.
+  //    Is that possible?
+  if (transformedResp.messages && Array.isArray(transformedResp.messages.value)) {
+    transformedResp.messages.value.forEach(message => {
+      message.message = getMessage(message);
+    });
+  }
   return transformedResp;
 };
 
-module.exports = uiSchemaLabelTransformer;
+export default uiSchemaLabelTransformer;
```

Comment:
should we just stick to named exports here instead of mixing both default and named exports?

Reply:
I have a mixing feeling. I used to like to stick with one style of exporting. But looks like when people starts to use ES6 import/export, it's likely get mixed because of the flexibility.

I establish mind set like
- `export default` creates the default behavior of this module
- `export` creates extra utilities.

Another reason is the usage other 2 `transformer` is using default import so I want to keep consistent.
`import xxxTransformer from './xxxTransformer'`
hence ended up using `export {getMessages}`

Reply:
@nikhilvenkatraman-okta @pradeepdewda-okta how about this?
```
export { uiSchemaLabelTransformer as default, getMessage };

```

Reply:
I don't mind that. Seems descriptive enough.